contrib/lkt/language/parser.py

```diff
@@ -1191,110 +1191,68 @@ def lexer():
     )
 
     @langkit_property()
-    def check_correctness_pre():
-        # All grammars must have an associated lexer. First check that the
-        # @with_lexer annotation is present.
+    def check_legality():
```

Comment:
It seems to me that in order to be able to resolve names inside a grammar definition, we need to be sure that the lexer has been correctly resolved first (because there can be references to tokens inside the grammar). Since this check is now done as a legality check, it seems to me that it is too late and some references in the grammar will fail just because the referenced lexer couldn't be resolved.

Reply:
You're right, good catch. I'll put back some or all of those checks in `check_pre`